src/trace_processor/perfetto_sql/intrinsics/functions/args.cc

```diff
@@ -0,0 +1,206 @@
+// Copyright (C) 2025 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/trace_processor/perfetto_sql/intrinsics/functions/args.h"
+
+#include <sstream>
+#include "src/trace_processor/sqlite/bindings/sqlite_result.h"
+#include "src/trace_processor/sqlite/bindings/sqlite_type.h"
+#include "src/trace_processor/util/args_utils.h"
+
+namespace perfetto::trace_processor {
+namespace {
+// We care about the argument order, so we can't use jsoncpp here.
+void WriteAsJson(const ArgNode& node,
+                 const TraceStorage* storage,
+                 std::ostringstream& ss) {
```

Comment:
Please never use string stream in stdlib. Use StringWriter or some other mechanism, stringstream is banned for binary size reasons.

Reply:
Is there a mechanism which dynamically allocates a string? An arg set can be quite large, and StringWriter only writes into a fixed-size strings, I believe.

Reply:
I guess not. But in that case, prefer creating a mechanism to do this (maybe built on top of string/string_view/stackstring instead of using stringstream).

Reply:
Ok, let me do this, then. I'll start with renaming `StringWriter` to `BufferedStringWriter` (as I was confused by the fact) and add `DynamicStringWriter`.